src/lib.rs

```diff
@@ -1,129 +1,3 @@
-        [0x02u8, 0x95, 0xb4, 0x3f] => Ok((Network::Bitcoin, "sh(wsh".to_string())), // Ypub
-        [0x04u8, 0xb2, 0x47, 0x46] => Ok((Network::Bitcoin, "wpkh".to_string())), // zpub
-        [0x02u8, 0xaa, 0x7e, 0xd3] => Ok((Network::Bitcoin, "wsh".to_string())), // Zpub
-        _ => Err(base58::Error::InvalidVersion(version.to_vec())),
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use crate::ElectrumExtendedPubKey;
-    use bitcoin::util::bip32::ChildNumber;
-    use miniscript::descriptor::DescriptorPublicKey;
-    use std::str::FromStr;
-
-    #[test]
-    fn test_vpub() {
-        let electrum_xpub = ElectrumExtendedPubKey::from_str("vpub5VXaSncXqxLbdmvrC4Y8z9CszPwuEscADoetWhfrxDFzPUbL5nbVtanYDkrVEutkv9n5A5aCcvRC9swbjDKgHjCZ2tAeae8VsBuPbS8KpXv").unwrap();
-        assert_eq!(electrum_xpub.xpub.to_string(),"tpubD9ZjaMn3rbP1cAVwJy6UcEjFfTLT7W6DbfHdS3Wn48meExtVfKmiH9meWCrSmE9qXLYbGcHC5LxLcdfLZTzwme23qAJoRzRhzbd68dHeyjp");
-        assert_eq!(electrum_xpub.kind, "wpkh");
-        let descriptors = electrum_xpub.to_descriptors();
-        assert_eq!(descriptors[0], "wpkh(tpubD9ZjaMn3rbP1cAVwJy6UcEjFfTLT7W6DbfHdS3Wn48meExtVfKmiH9meWCrSmE9qXLYbGcHC5LxLcdfLZTzwme23qAJoRzRhzbd68dHeyjp/0/*)");
-        assert_eq!(descriptors[1], "wpkh(tpubD9ZjaMn3rbP1cAVwJy6UcEjFfTLT7W6DbfHdS3Wn48meExtVfKmiH9meWCrSmE9qXLYbGcHC5LxLcdfLZTzwme23qAJoRzRhzbd68dHeyjp/1/*)");
-        let xpub = electrum_xpub.xpub();
-        assert_eq!(xpub.to_string(), "tpubD9ZjaMn3rbP1cAVwJy6UcEjFfTLT7W6DbfHdS3Wn48meExtVfKmiH9meWCrSmE9qXLYbGcHC5LxLcdfLZTzwme23qAJoRzRhzbd68dHeyjp");
-    }
-
-    #[test]
-    fn test_slip121_vectors() {
-        // from https://github.com/satoshilabs/slips/blob/master/slip-0132.md
-        test_first_address("xpub6BosfCnifzxcFwrSzQiqu2DBVTshkCXacvNsWGYJVVhhawA7d4R5WSWGFNbi8Aw6ZRc1brxMyWMzG3DSSSSoekkudhUd9yLb6qx39T9nMdj","1LqBGSKuX5yYUonjxT5qGfpUsXKYYWeabA");
-        test_first_address("ypub6Ww3ibxVfGzLrAH1PNcjyAWenMTbbAosGNB6VvmSEgytSER9azLDWCxoJwW7Ke7icmizBMXrzBx9979FfaHxHcrArf3zbeJJJUZPf663zsP","37VucYSaXLCAsxYyAPfbSi9eh4iEcbShgf");
-        test_first_address("zpub6rFR7y4Q2AijBEqTUquhVz398htDFrtymD9xYYfG1m4wAcvPhXNfE3EfH1r1ADqtfSdVCToUG868RvUUkgDKf31mGDtKsAYz2oz2AGutZYs","bc1qcr8te4kr609gcawutmrza0j4xv80jy8z306fyu");
-    }
-
-    fn test_first_address(electrum_xpub: &str, expected_first_address: &str) {
-        let electrum_xpub = ElectrumExtendedPubKey::from_str(electrum_xpub).unwrap();
-        let descriptors = electrum_xpub.to_descriptors();
-        let descriptor: miniscript::Descriptor<DescriptorPublicKey> =
-            descriptors[0].parse().unwrap();
-        let first_address = descriptor
-            .derive(ChildNumber::from_normal_idx(0).unwrap())
-            .address(electrum_xpub.xpub.network)
-            .unwrap()
-            .to_string();
-        assert_eq!(expected_first_address, first_address);
-    }
-}
+pub mod electrumextendedkey;
+pub mod electrumextendedprivkey;
+pub mod electrumextendedpubkey;
```

Comment:
can we add a re-export of `ElectrumExtendedPrivKey` and `ElectrumExtendedPubKey` so that import in bin becomes just

`use libelectrum2descriptors::{ElectrumExtendedPrivKey,lectrumExtendedPubKey}`